pkg/syncer/upsync/upsync_reconcile.go

```diff
@@ -0,0 +1,261 @@
+			return false, err
+		}
+
+		// Status also needs to be upsynced so let's do it in 3 steps:
+		// - create the resource
+		createdResource, err := c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).Create(ctx, preparedResource, metav1.CreateOptions{})
+		if err != nil {
+			return false, err
+		}
+		// - update the status as a distinct action,
+		preparedResource.SetResourceVersion(createdResource.GetResourceVersion())
+		updatedResource, err := c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).UpdateStatus(ctx, preparedResource, metav1.UpdateOptions{})
+		if err != nil {
+			return false, err
+		}
+		// - finally update the main content again to set the resource version annotation to the value of the downstream resource version
+		preparedResource.SetAnnotations(addResourceVersionAnnotation(resourceVersionDownstream, preparedResource.GetAnnotations()))
+		preparedResource.SetResourceVersion(updatedResource.GetResourceVersion())
+		_, err = c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).Update(ctx, preparedResource, metav1.UpdateOptions{})
+		return false, err
+	}
+
+	resourceVersionUpstream := upstreamObject.GetAnnotations()[ResourceVersionAnnotation]
+	if resourceVersionDownstream != resourceVersionUpstream {
+		// Resource version annotation on the upstream resource doesn't match the downstream resource version
+		// => let's update the resource upstream.
+		logger.Info("Updating upstream resource")
+		preparedResource := c.prepareResourceForUpstream(ctx, gvr, upstreamNamespace, upstreamClusterName, downstreamResource)
+		if err != nil {
+			return false, err
+		}
+
+		if !dirtyStatus {
+			// if no status needs to be upsynced upstream, then we can set the resource version annotation to the downstream resource version
+			// at the same time as we update the main resource content.
+
+			preparedResource.SetAnnotations(addResourceVersionAnnotation(resourceVersionDownstream, preparedResource.GetAnnotations()))
+			existingResource, err := c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).Get(ctx, preparedResource.GetName(), metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			preparedResource.SetResourceVersion(existingResource.GetResourceVersion())
+			_, err = c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).Update(ctx, preparedResource, metav1.UpdateOptions{})
+			// If the downstream resource is marked for deletion, let's requeue it to manage the deletion timestamp
+			return markedForDeletionDownstream, err
+		}
+
+		// Status also needs to be upsynced so let's do it in 3 steps:
+		// - update the upstream resource main content
+
```

Comment:
```suggestion
		// slow path: status changed => we need 3 steps
		// 1. update main resource
```

Reply:
Fixed in commit fffff6927c99c86702c7709881a77057331950cd